src/gui/FileBrowser.cpp

```diff
@@ -703,6 +752,25 @@ QList<QAction*> FileBrowserTreeWidget::getContextActions(FileItem* file, bool so
 	return result;
 }
 
+void FileBrowserTreeWidget::addFavorite(QString item)
+{
+	item = ConfigManager::removeTrailingSeparators(item);
+	ConfigManager::addFavoriteItem(item);
+	FileBrowser::getFavoritesBrowserInstance()->reloadTree();
```

Comment:
Reloading the entire tree just to update the tree can be inefficient. The coupling by storing a reference to the "Favorites" browser can also be avoided. I would recommend the following:

1. If the browser is the "Favorites" browser, add the item using `addTopLevelItem`.
2. If not, do nothing.

That is why I wouldn't maintain a separate list of the items, but update the tree dynamically.